aqua-organisation/ah-django-Aqua
# #165177127 Users should be able to edit their articles

#### What does this PR do?
* Add an endpoint to update an article
#### Description of Task to be completed?
* Ensure only authenticated users edit articles
* Ensure only users who created an article can edit it
#### How should this be manually tested?
* After cloning this repo, cd into it and switch to this branch
* Start the development server using ```python manage.py runserver```

Given an existing article such as below
```
{
    "article": {
        "slug": "lorem-ipsum-dolor-old",
        "title": "Lorem ipsum dolor old",
        "description": "Lorem ipsum dolor sit amet, consectetur",
        "body": "Ask Question I added validate_mail and also installed",
        "createdAt": "2019-04-12 07:54:03.437416+00:00",
        "updatedAt": "2019-04-12 16:46:32.699055+00:00",
        "read_time": "1 min read",
        "favoritesCount": 0,
        "author": {
            "username": "reifred",
            "bio": "",
            "image": ""
        }
    }
}
```
* Using Postman test the endpoint below

Update using an article slug name in this case.
```
PATCH '/api/articles/lorem-ipsum-dolor-old/'
Body Request.
{
	"article":{
		"body": "blog body for post 1 changed"
	}
}
Expected Output
{
    "article": {
        "slug": "lorem-ipsum-dolor-old",
        "title": "Lorem ipsum dolor old",
        "description": "Lorem ipsum dolor sit amet, consectetur",
        "body": "blog body for post 1 changed",
        "createdAt": "2019-04-12 07:54:03.437416+00:00",
        "updatedAt": "2019-04-12 16:46:32.699055+00:00",
        "read_time": "1 min read",
        "favoritesCount": 0,
        "author": {
            "username": "reifred",
            "bio": "",
            "image": ""
        }
    }
}
```
In case an article slug given doesnot exist.
```
PATCH '/api/articles/blog-post-2/'
Request Body.
{
	"article":{
		"body": "blog body for post 1 changed"
	}
}
Expected Output
{
    "errors": {
        "detail": "Article does not exist in your articles."
    }
}
```
Error expected in case an author tries to edit another authors' article.
```
{
    "errors": {
        "detail": "You do not have permission to perform this action."
    }
}
```
#### What are the relevant pivotal tracker stories?
[#165177127](https://www.pivotaltracker.com/story/show/165177127)
